.github/workflows/test.yml

```diff
@@ -33,7 +33,42 @@ jobs:
     - uses: ruby/setup-ruby@v1
       with:
         ruby-version: ${{ matrix.ruby }}
-        bundler: 1.17.3
+        bundler: 2.2.29
+        bundler-cache: true
+    - run: RAILS_ENV=test bundle exec rake db:create
+    - run: bundle exec rspec
+
+  test-ruby-3:
+    name: Ruby 3 Unit Tests
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_DATABASE: activerecord_mysql_enum_test
+          MYSQL_USER: user
+          MYSQL_PASSWORD: password
+          MYSQL_ROOT_PASSWORD: rootpassword
+        ports:
+        - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['3.0', 3.1]
+        gemfile:
+        - gemfiles/rails_6.gemfile
```

Comment:
There's no need to split between ruby 2 and ruby 3 like this just to exclude testing of rails 5 on ruby 3+. The `matrix` syntax allows the ability to do an `exclude` section so the matrix definition would look like the following:

```
matrix:
  ruby: [2.5, 2.6, 2.7, '3.0', 3.1]
  gemfile:
  - Gemfile
  - gemfiles/rails_5.gemfile
  - gemfiles/rails_6.gemfile
  exclude:
  - ruby: '3.0'
    gemfile: gemfiles/rails_5.gemfile
  - ruby: 3.1
    gemfile: gemfiles/rails_5.gemfile
```